iml-gui/crate/src/components/action_dropdown.rs

```diff
@@ -0,0 +1,293 @@
+use crate::{
+    components::{dropdown, font_awesome, modal, tooltip, Placement},
+    generated::css_classes::C,
+    GMsg, MergeAttrs as _, WatchState,
+};
+use futures::{
+    channel::oneshot,
+    future::{self, Either},
+    Future, FutureExt,
+};
+use gloo_timers::future::TimeoutFuture;
+use iml_wire_types::{ApiList, AvailableAction, CompositeId, Label, ToCompositeId};
+use seed::{prelude::*, *};
+use std::sync::Arc;
+
+pub type AvailableActions = ApiList<AvailableAction>;
+
+pub trait ActionRecord: ToCompositeId + serde::Serialize + Label + Clone {}
+impl<T: ToCompositeId + serde::Serialize + Label + Clone> ActionRecord for T {}
+
+pub fn composite_ids_to_query_string(xs: &[CompositeId]) -> String {
+    let mut xs: Vec<String> = xs
+        .iter()
+        .map(|x| format!("composite_ids={}", x))
```

Comment:
This is carryover from iml-wasm-components. There is probably a reason why this was sorted, but I need to check...

Reply:
The reason why this was done is because there was no go-between for records into composite ids. 

This approach was taken so that the api response could be zipped with the returned Id's here.

Now that `ErasedRecord` has been created, this is no longer needed and the sort has been removed.